sycl/include/CL/sycl/multi_ptr.hpp

```diff
@@ -358,11 +382,18 @@ template <access::address_space Space> class multi_ptr<void, Space> {
           Accessor)
       : multi_ptr(Accessor.get_pointer()) {}
 
+#if defined(RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR)
+  using ReturnPtr =
+      typename std::conditional<Space == access::address_space::constant_space,
+                                const void *, void *>::type;
+#else
+  using ReturnPtr = void *;
```

Comment:
Maybe `VoidPtr` or `ReturnVoidPtr` instead? (Minor, optional, feel free to ignore)

Reply:
Not sure about it. This is specialization of multi_ptr class from void type, I think `ReturnPtr ` fits here.